src/token/aft34/extensions/payable_mint.rs

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) 2022-2023 Allfeat labs
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the"Software"),
+// to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+pub use crate::{
+    aft34,
+    traits::aft34::{extensions::metadata::*, extensions::payable_mint::*, *},
+};
+pub use aft34::{
+    AFT34Impl, BalancesManager as _, Internal as _, InternalImpl as _, Operator, Owner,
+};
+use ink::prelude::string::{String, ToString};
+use openbrush::{
+    contracts::ownable::*,
+    traits::{AccountId, Balance, Storage},
+};
+
+use super::metadata::AFT34MetadataImpl;
+
+#[derive(Default, Debug)]
+#[openbrush::storage_item]
+pub struct Data {
+    pub last_token_id: u64,
+    pub max_supply: u64,
+    pub price_per_mint: Balance,
+}
+
+#[openbrush::trait_definition]
+pub trait AFT34PayableMintImpl:
+    aft34::Internal
+    + aft34::aft34::AFT34Impl
+    + aft34::extensions::metadata::Internal
+    + aft34::extensions::payable_mint::Internal
+    + aft34::extensions::metadata::AFT34MetadataImpl
+    + aft34::InternalImpl
+    + Storage<Data>
+    + Storage<ownable::Data>
+{
+    fn mint(&mut self, to: AccountId, mint_amount: u64) -> Result<(), AFT34Error> {
+        self.check_value(Self::env().transferred_value(), mint_amount)?;
+        self.check_amount(mint_amount)?;
+
+        let next_to_mint = self.data::<Data>().last_token_id + 1; // first mint id is 1
+        let mint_offset = next_to_mint + mint_amount;
+
+        for mint_id in next_to_mint..mint_offset {
+            aft34::InternalImpl::_mint_to(self, to, Id::U64(mint_id))?;
```

Comment:
can be replaced with:

```suggestion
            self._mint_to(to, Id::U64(mint_id))?;
```